### DISCUSSION TITLE
Using application without need for authentication

### DISCUSSION THREAD

**Discussion by adizag** [gravitational/teleport#31458] [Q&A] [answered]
Hi,
I know this is not the intention of Teleport but I still need it for a specific aim.

I want the application address to bypass the login to Teleport step, so everyone using the address will get straight to the destination.

Do you know if it can be done and how?

I really appreciate any help you can provide.

**Comment by webvictim** [comment, marked as answer]
This can't be done. Teleport enforces that a user must have a valid certificate before it allows any traffic to flow to the application, so it's not possible to allow "anonymous" access to applications behind Teleport.

It sounds like what you want is a standard reverse proxy - something like nginx, Apache, Caddy etc?

**Reply by adizag** [reply]
I do, but since my ISP is providing Double NAT network structure I can't have access to my server without some tunnel...
For some services I use Cloudflare, but for this service, I can't use Cloudflare because it is for bigger capacity which the free Cloudflare account doesn't allow.

Any idea for another service I could use that could provide me with a tunnel while allowing anonymous login (and free?)?

**Reply by webvictim** [reply]
Do you mean CGNAT? They don't give you a public IP? Usually with ISPs that do this you can pay them $5/month or equivalent to get a static IP.

The alternative I see used often is to buy a $5/month VPS and establish a persistent reverse SSH port forward from your server -> VPS so that traffic can flow back into your network over that tunnel.
